### PR TITLE
Add Syntax Themes

### LINKED ISSUE #59: Code snippets unreadable in default (Atom.io) theme
### Thanks in advance for your bug report!

- [X] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [X] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [X] Have you checked you are on the latest release of Pulsar?

### What happened?

The colors in code snippets are broken in the default theme (Atom.io).

### Pulsar version

Website

### Which OS does this happen on?

🐧 Arch based (Manjaro, Garuda, etc.)

### OS details

Doesn't matter I believe

### Which CPU architecture are you running this on?

None

### What steps are needed to reproduce this?

1. Go to a package with code block in README.

### Additional Information:

image

image

https://web.pulsar-edit.dev/packages/x-terminal

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR uses the many themes available from `Highlight.js` to properly have the code blocks within a packages ReadMe match the theme of the page selected by the user.

This PR also adds these new steps to the docs on creating your own theme. And retroactively applies the most properly matching syntax theme to all created themes.

---

Closes #59

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,10 +22,18 @@ If there isn't a theme yet that fits you, creating a new theme is welcome, encou
 
 First create the theme itself, by laying out the theme variables in `./src/site.css`. The top of this file should have every other supported theme listed along with the variables that can and will need to be modified for your theme.
 
-Once the theme is created, ensure to add it as an option in `./public/site.js` within the function `changeTheme` and add a new check in the switch statement. Setting the body to have your theme as an attribute value, and saving the new theme option into the users local storage.
+After creating the colours of your theme you'll need to find what syntax theme matches your new theme the best. The Pulsar Package Frontend uses [`Highlight.js`](https://highlightjs.org/) to handle highlighting code blocks that will appear within packages ReadMes, so you can browse all their available themes on their [website](https://highlightjs.org/static/demo/). Once you've found the one you like, make sure to jot down it's CND URL.
```
Comment (Daeraxa) [inline README.md:25]:
```suggestion
After creating the colors of your theme you'll need to find what syntax theme matches your new theme the best. The Pulsar Package Frontend uses [`Highlight.js`](https://highlightjs.org/) to handle highlighting code blocks that will appear within package READMEs, so you can browse all their available themes on their [website](https://highlightjs.org/static/demo/). Once you've found the one you like, make sure to jot down its CND URL.
```

As much as I love the correct British spelling I feel this should be in American English :P
Also I think "ReadMes" looks a bit odd and it is usually capitalised.

File: README.md [hunk 2]
```diff
@@ -22,10 +22,18 @@ If there isn't a theme yet that fits you, creating a new theme is welcome, encou
...
+
+Once the theme is created, ensure to add it as an option in `./public/site.js` within the function `changeTheme` and add a new check in the switch statement. Setting the body to have your theme as an attribute value, and saving the new theme option into the users local storage. Additionally in your new section of the switch statement ensure to add `changeSyntax("Your new Syntax Theme Name");`. This ensures the syntax is changed at the same time as your theme. Then within the function `changeSyntax()` add your themes name to the `syntaxes` array.
```
Comment (Daeraxa) [inline README.md:27]:
```suggestion
Once the theme is created, ensure to add it as an option in `./public/site.js` within the function `changeTheme` and add a new check in the switch statement. Setting the body to have your theme as an attribute value, and saving the new theme option into the users local storage. Additionally in your new section of the switch statement ensure to add `changeSyntax("Your new Syntax Theme Name");`. This ensures the syntax is changed at the same time as your theme. Then within the function `changeSyntax()` add your theme's name to the `syntaxes` array.
```

File: README.md (hunk 1 above)
Comment (confused-Techie) [reply README.md:25]: Always does take me a second to realize why my spell check gets frustrated at `colours` but otherwise looks great thanks!

**Comment by confused-Techie** [comment]
@Daeraxa done and done

**Review by Daeraxa — COMMENTED** [review]
Just a few comments for the docs side of things.

**Comment by savetheclocktower** [comment]
Not a blocker for this PR, but I think it would be nice to eventually load the syntax themes dynamically, instead of loading all of them (four now, probably more later) on page load. I never even noticed that the frontend offered different themes until today, and I'm guessing lots of people will never change the theme at all, so it'd be good to avoid sending them CSS they'll never use.

**Comment by confused-Techie** [comment]
> Not a blocker for this PR, but I think it would be nice to eventually load the syntax themes dynamically, instead of loading all of them (four now, probably more later) on page load. I never even noticed that the frontend offered different themes until today, and I'm guessing lots of people will never change the theme at all, so it'd be good to avoid sending them CSS they'll never use.

This is a very good point. Plus noting that the themes aren't super noticeable might be a good idea to choose a better default then. Since honestly I dislike the default the most (As it's based on the original Atom.io website)

But yeah might be added in an additional PR, but if this is merged without that feature I'll make an issue for it so I don't forget

**Comment by confused-Techie** [comment]
@savetheclocktower Alright, so I did go ahead and end up add dynamic Syntax Theme CSS loading to this PR.

So now during the first page load you will always download the default syntax theme. But if a theme is set than the JavaScript will remove the previous load and add an updated one to the correct URL.

This does mean at worst, when someone has a custom theme set they will initially download the page along with the default syntax theme, then will go ahead and download the custom specified syntax theme right afterwards.
That does mean at most you will download two syntax themes, and will possibly reduce our CLS score, but otherwise (loading wise) is still in improvement over downloading all syntax themes right away.

File: src/site.css [hunk 3]
```diff
@@ -128,10 +128,6 @@ nav.active {
   @apply inline-block mr-2 text-text fill-primary;
 }
 
-#readme .hljs {
-  @apply text-text bg-secondary;
```
Comment (savetheclocktower) [inline src/site.css:132]: Does this CSS get pre-processed? I wasn't aware that `@apply` was being used anywhere — it was [proposed but withdrawn](https://www.xanthir.com/b4o00).

Comment (savetheclocktower) [reply src/site.css:132]: Ah, it's Tailwind. (*\*shudder\**) Never mind!

**Review by savetheclocktower — COMMENTED** [review]
(disregard; accidentally clicked “start a review”)

File: src/site.css (hunk 3 above)
Comment (confused-Techie) [reply src/site.css:132]: Yes this CSS file is processed before being used in production (or dev for that matter).

We use `tailwind` to process our CSS file, and `@apply` is a directive within `tailwind`

You can look in our `package.json` to see the ` "postInstall": "npm run tailwind:build && npm run mini.js && npm run mini:css"`

Those above steps use `tailwind` to process the CSS file and output it to `./public/site.css` then the other two steps both use `minify` to minify our production JavaScript and CSS into `./public/site.min.js` and `./public/site.min.css` respectively

**Pull request merged**